tests/test_components.py

```diff
@@ -30,3 +30,45 @@ def test_serialization(self):
         edge_dict = edge.to_dict()
         self.assertDictEqual(edge_dict['source'], source.to_dict())
         self.assertDictEqual(edge_dict['destination'], destination.to_dict())
+
+        # test serialization with dicts
+        edge1 = Edge.from_dict(edge_dict)
+        self.assertEqual(edge, edge1)
+
+        # test edge with mixed nodes
+        source = Node(identifier='a', meta={'color': 'blue'}, variable_type=NodeVariableType.BINARY)
+        destination = TimeSeriesNode(identifier='b', meta={'color': 'green'}, variable_type=NodeVariableType.CONTINUOUS)
+        edge = Edge(
+            source=source, destination=destination, edge_type=EdgeType.UNKNOWN_DIRECTED_EDGE, meta={'color': 'red'}
+        )
+
+        # check node class is preserved
+        edge2 = Edge.from_dict(edge.to_dict())
+        self.assertEqual(edge, edge2)
+
+        # test node serialization-deserialization
+        node = Node(identifier='a', meta={'color': 'blue'}, variable_type=NodeVariableType.BINARY)
+        node1 = Node.from_dict(node.to_dict())
+
+        self.assertEqual(node, node1)
```

Comment:
meta is not included in equality check. Just the identifier. So would check meta and variable type are also equal.